File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {

        unordered_set<int> set;

        for (auto& num : nums) {
            auto pair = set.insert(num);
            if (!pair.second) {
                return true;
            }
        }

        return false;
        
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # s = set(nums)
        # return len(s) != len(nums)

        seen = set()

        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        
        return False



        